lib/services/access_list.go

```diff
@@ -276,9 +418,21 @@ func (a AccessListMembershipChecker) IsAccessListMember(ctx context.Context, ide
 	}
 
 	member, err := a.members.GetAccessListMember(ctx, accessList.GetName(), username)
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
 	if trace.IsNotFound(err) {
-		// The member has not been found, so we know they're not a member of this list.
-		return trace.NotFound("user %s is not a member of the access list", username)
+		// try find if the user could be a member of any lists by recursing
+		err := a.recursiveIsAccessListMemberCheck(ctx, identity, accessList)
+		if trace.IsNotFound(err) {
+			// The member has not been found, so we know they're not a member of this list.
+			return trace.NotFound("user %s is not a member of the access list", username)
+		}
+		if err != nil {
+			// Some other error has occurred
```

Comment:
nit: The comment doesn't carry to much value for the reader can we remove it or explain the error flow ?